cirrus-ci_retrospective/bin/cirrus-ci_retrospective.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+set -eo pipefail
+
+# Execute inside a github action, using a completed check_suite event's JSON file
+# as input.  Queries details about the concluded Cirrus-CI build, tasks, artifacts,
+# execution environment, and associated repository state.
+
+source $(dirname "${BASH_SOURCE[0]}")/../lib/$(basename "${BASH_SOURCE[0]}")
+
+if ((DEBUG)); then
+    dbg "# Warning: Debug mode enabled:  NOT cleaning up '$TMPDIR' upon exit."
+else
+    trap "rm -rf $TMPDIR" EXIT
+fi
+
+verify_env_vars
+
+INTERMEDIATE_OUTPUT_EXT=".json_item"
+OUTPUT_JSON_FILE="$GITHUB_WORKSPACE/${SCRIPT_FILENAME%.sh}.json"
+
+# Confirm expected triggering event
+[[ "$(jq --slurp --compact-output --raw-output '.[0].action' < $GITHUB_EVENT_PATH)" == "completed" ]] || \
+    die "Expecting github action event action to be 'completed'"
```

Comment:
Messages like this, that don't include the actual value, are impossible to debug on failure. Could I encourage you to refactor this such that `.[0].action` is stored in a temporary variable, and the `die` message includes it? Same with line 31 below

Reply:
okay, I fixed this and several others.  Thx for the suggestion.

Reply:
Thank you. I keep jumping in and out of this one, mostly out due to distractions. I'm willing to have you merge it and see how it works in action -- so much of it is far above my head.

Reply:
> so much of it is far above my head.

Mine too actually :confused:, most of what I've got is the result of countless tweak-push-observe iterations over on the **automation_sandbox** repo.  As I mentioned before, all the GraphQL I worked up using graphical tools, since it's too dense to grasp at the individual "node" level, you need to "see" how it's all connected to write effective queries/mutations.

Github Actions is by far the most frustrating aspect.  It's chock-full of useless features and unnecessary complexities, and bound together with less-than-helpful documentation.  Huge distractions and dead-end rat-holes basically :confounded: I can't wait to get back to the simpler world of Cirrus-CI :grinning: